lib/scope.js

```diff
@@ -331,13 +332,26 @@ module.exports = {
       let load_result = await scope.getLoadData( scope, { timeout: attempt_timeout });
 
       if ( load_result.error ) {
-        await scope.addToReport(scope, { type: `error`, value: load_result.error });
-        expect( load_result.error ).to.equal( '' );
-      }
+
+        // `ModuleNotFoundError` could just be the server trying to reload, so give
+        // it more time until the timeout is almost over. End early to give good message.
+        // See https://github.com/plocket/docassemble-cucumber/issues/367
+        let global_timeout_is_almost_done = (Date.now() - start_time) >= (scope.timeout - 100);
+        if ( load_result.error.includes( `ModuleNotFoundError` ) && !global_timeout_is_almost_done ) {
+          await scope.addToReport(scope, { type: `warning`, value: `Attempted to load module. Got "${ load_result.error }"` });
+          await scope.load( scope, file_name, attempt_num, start_time );
+
+        } else {
+          // Other errors, or reaching timeout, causes an excpetion and is shown to the developer
+          await scope.addToReport(scope, { type: `error`, value: load_result.error });
+          expect( load_result.error ).to.equal( '' );
+        }
+
+      }  // ends if load_result.error
 
     } catch ( err ) {
       // If it doesn't work, maybe try again
-      if ( attempt_num <= max_attempts ) { await scope.load( scope, file_name, attempt_num ); }
+      if ( attempt_num <= max_attempts ) { await scope.load( scope, file_name, attempt_num, start_time ); }
```

Comment:
Changing this to `if ( attempt_num <= max_attempts ) || global_timeout_is_almost_done` would clarify the logic a little bit. That way, all the logging could be in the `if` block above, and the retrying in the `catch` block here. (Doing this will require changing `global_timeout_is_almost_done` to also take into account `load_result.error.includes( 'ModuleNotFoundError' )`.

Reply:
My concern is that we'll trigger an error before the check and that we still want to give it another try or two. That means that this logic needs to be in here as well. We do need to check for almost done here too, though, so I might move that up a level and use it in both places. How does that sound?

Reply:
I see the new issue. That proposed resolution sounds good to me.